File: task_equivalents/llm_translations/llama/optimized/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd
from sklearn.preprocessing import scale

# Load data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?', index_col=0)

# Drop rows with missing values
auto = auto.dropna()

# Create a binary column 'mpg01'
auto['mpg01'] = (auto['mpg'] > auto['mpg'].median()).astype(int)

# Drop 'mpg' and 'name' columns
auto = auto.drop(columns=['mpg', 'name'])

# Define columns to scale and not to scale
var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']

# Scale continuous variables
auto[var_to_scale] = scale(auto[var_to_scale])

# Split data into features and target
x = auto.drop(columns='mpg01')
y = auto['mpg01']

File: task_equivalents/llm_translations/llama/optimized/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(readr)
library(dplyr)

Auto <- read_csv('data/csv/Auto.csv', na = c("?")) %>%
  drop_na() %>%
  mutate(
    mpg01 = ifelse(mpg > median(mpg), 1, 0)
  ) %>%
  select(-c(mpg, name))

var_no_scale <- c('cylinders', 'year', 'origin', 'mpg01')
var_to_scale <- c('displacement', 'horsepower', 'weight', 'acceleration')

scaled_var <- Auto %>%
  select(all_of(var_to_scale)) %>%
  scale() %>%
  as.data.frame()

df <- bind_cols(scaled_var, Auto %>% select(all_of(var_no_scale)))

x <- df %>% select(-mpg01)
y <- df$mpg01